clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3809,8 +3811,53 @@ private GenericRecord readStoredValueRecord(
    */
   private void restoreProducerStatesForLeaderConsumption(int partition) {
     consumerDiv.clearPartition(partition);
-    cloneProducerStates(partition, consumerDiv);
-    // TODO: the Global RT DIV version needs to be implemented in a future PR
+    if (isGlobalRtDivEnabled()) {
+      loadGlobalRtDiv(partition);
+    } else {
+      cloneProducerStates(partition, consumerDiv);
+    }
+  }
+
+  private void loadGlobalRtDiv(int partition) {
+    kafkaClusterIdToUrlMap.forEach((clusterId, brokerUrl) -> {
+      loadGlobalRtDiv(partition, brokerUrl);
+    });
+  }
+
+  /**
+   * Load the stored Global RT DIV object from the storage engine into the Consumer DIV during state transition
+   * to LEADER. The RT DIV needs to be loaded per-broker url, and the latest consumed RT offset needs to be updated.
+   */
+  private void loadGlobalRtDiv(int partition, String brokerUrl) {
+    PartitionConsumptionState pcs = partitionConsumptionStateMap.get(partition);
+    final PubSubTopic topic = pcs.getOffsetRecord().getLeaderTopic(pubSubTopicRepository);
+    final PubSubTopicPartition topicPartition = new PubSubTopicPartitionImpl(topic, pcs.getPartition());
+
+    String globalRtDivKey = getGlobalRtDivKeyName(brokerUrl);
+    byte[] keyBytes = globalRtDivKey.getBytes();
+    final ChunkedValueManifestContainer valueManifestContainer = new ChunkedValueManifestContainer();
+    GenericRecord valueRecord = readStoredValueRecord(
+        pcs,
+        keyBytes,
+        AvroProtocolDefinition.GLOBAL_RT_DIV_STATE.getCurrentProtocolVersion(),
+        topicPartition,
+        valueManifestContainer);
+    if (valueRecord == null) {
+      return; // it may not exist (e.g. this is the first leader to be elected)
+    }
+
+    Object value = valueRecord.get(globalRtDivKey);
+    if (value instanceof GlobalRtDivState) {
+      GlobalRtDivState globalRtDivState = (GlobalRtDivState) value;
+      final Map<CharSequence, ProducerPartitionState> producerStates = globalRtDivState.getProducerStates();
+      PartitionTracker.TopicType realTimeTopicType = PartitionTracker.TopicType.of(REALTIME_TOPIC_TYPE, brokerUrl);
+      consumerDiv.setPartitionState(realTimeTopicType, pcs.getPartition(), producerStates);
+      final long latestConsumedRtOffset = globalRtDivState.getLatestOffset(); // LCRO
+      pcs.updateLatestConsumedRtOffset(brokerUrl, latestConsumedRtOffset);
+    } else {
+      LOGGER.warn("Unable to load Global RT DIV: {}", globalRtDivKey);
```

Comment:
1. Let's also include store, version, and partition in the log for debugging.
2. This is never expected to happen in real, but in case it happens. My thoughts is that we should log enough information for debugging and then move on. Yeah, We lose all global div infos in the past and treat it as a brand new beginning.